spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/authentication/MicroserviceInstanceService.java

```diff
@@ -0,0 +1,33 @@
+/*
+
+ * Copyright (C) 2020-2022 Huawei Technologies Co., Ltd. All rights reserved.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.huaweicloud.governance.authentication;
+
+/**
+ * microservice instance service
+ */
+public interface MicroserviceInstanceService {
+    void setPublickey(String publicKeyEncoded);
+
+    String getInstanceId();
+
+    String getServiceId();
+
+    Object getCustomerMicroservice(String serviceId, String instanceId);
+
+    String getPublicKeyFromInstance(String instanceId, String serviceId);
```

Comment:
Maybe can think better extensions . Let service comb and nacos has minimal code

Reply:
Have change achieve